tests/tests/tier0/bluechi-change-port-with-p-cmd-option/test_bluechi_change_port_with_p_cmd_option.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from typing import Dict
+import os
+import time
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+
+NODE_FOO = "node-foo"
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    node_foo = nodes[NODE_FOO]
+    bluechi_agent_str = "bluechi-agent"
+    bluechi_controller_str = "bluechi-controller"
+
+    # change controller port to 8421
+    ctrl.restart_with_modified_service_file(
+        bluechi_controller_str,
+        [
+            f"sed -i '/^ExecStart=/ s/$/ -p 8421/'"
+        ],
+    )
+    assert ctrl.wait_for_unit_state_to_be(bluechi_controller_str, "active")
+
+    # Check if port 8421 is listenning
+    time.sleep(1)  # Give some time to the controller to start listening
+    _, pid = ctrl.exec_run("lsof -t -i:8421 -c 0")
+    _, output = ctrl.exec_run(f"ps -ocmd --no-headers {str(pid).strip()}")
+    assert "bluechi-c" in str(output)
+
+    # Check if node disconnected
+    result, _ = ctrl.run_python(os.path.join("python", "is_node_connected.py"))
+    assert result
+
+    # change node port to 8421
+    node_foo.restart_with_modified_service_file(
+        bluechi_agent_str,
+        [
+            f"sed -i '/^ExecStart=/ s/$/ -p 8421/'"
+        ],
+    )
+    assert node_foo.wait_for_unit_state_to_be(bluechi_agent_str, "active")
+
+    # Check if node connected
+    time.sleep(1)
+    _, pid = node_foo.exec_run("lsof -t -i:8421 -c 0")
+    _, output = node_foo.exec_run(f"ps -ocmd --no-headers {str(pid).strip()}")
+    assert "bluechi-a" in str(output)
```

Comment:
It would be great to reuse the cmd on port script added in https://github.com/eclipse-bluechi/bluechi/pull/845

Reply:
done, updated